### ISSUE TITLE
server live migration does not work

### ISSUE THREAD

**Issue by jcannava** [rcbops/ansible-lxc-rpc#6]
On the IAD labs, when migrating a running instance, the migration target enters an error state

Excerpt from nova-compute.log on the compute host (including stacktrace):
https://gist.github.com/jcourtois/8deb012fda83c4920411

I think this is the necessary fix:
https://bugzilla.redhat.com/show_bug.cgi?id=975014

**Comment by jacobwagner** [comment]
reopening, the authorized key file is not getting laid down on the computes

**Comment by jacobwagner** [comment]
more info

```
(.venv)root@569037-ctrl01:~/ansible-lxc-rpc/rpc_deployment# ll ~/.ssh/
authorized_keys  id_rsa           id_rsa.pub       known_hosts      nova             nova.pub         
```

```
TASK: [nova_compute_sshkey_setup | Set nova users shell to /bin/bash and generate ssh_key] *** 
changed: [compute1]
changed: [compute5]
changed: [compute4]
changed: [compute3]
changed: [compute8]
changed: [compute2]
changed: [compute7]
changed: [compute6]
changed: [compute10]
changed: [compute11]
changed: [compute9]
changed: [compute13]
changed: [compute12]
changed: [compute16]
changed: [compute18]
changed: [compute15]
changed: [compute14]
changed: [compute17]

TASK: [nova_compute_sshkey_setup | Copy private key up to nova nodes] ********* 
ok: [compute5]
ok: [compute7]
ok: [compute1]
ok: [compute4]
ok: [compute3]
ok: [compute2]
ok: [compute6]
ok: [compute10]
ok: [compute9]
ok: [compute11]
ok: [compute13]
ok: [compute12]
ok: [compute8]
ok: [compute18]
ok: [compute16]
ok: [compute15]
ok: [compute14]
ok: [compute17]

TASK: [nova_compute_sshkey_setup | Copy public key up to nova nodes] ********** 
ok: [compute1]
ok: [compute5]
ok: [compute4]
ok: [compute3]
ok: [compute2]
ok: [compute6]
ok: [compute11]
ok: [compute9]
ok: [compute13]
ok: [compute12]
ok: [compute8]
ok: [compute7]
ok: [compute16]
ok: [compute15]
ok: [compute14]
ok: [compute17]
ok: [compute10]
ok: [compute18]

TASK: [nova_compute_sshkey_setup | Add key to authorized_keys file] *********** 
ok: [compute2]
ok: [compute3]
ok: [compute5]
ok: [compute6]
ok: [compute9]
ok: [compute11]
ok: [compute8]
ok: [compute13]
ok: [compute1]
ok: [compute12]
ok: [compute4]
ok: [compute14]
ok: [compute17]
ok: [compute7]
ok: [compute10]
ok: [compute16]
ok: [compute15]
ok: [compute18]

TASK: [nova_compute_sshkey_setup | Prevent known_hosts from causing an issue] *** 
ok: [compute5]
ok: [compute1]
ok: [compute4]
ok: [compute3]
ok: [compute2]
ok: [compute6]
ok: [compute11]
ok: [compute9]
ok: [compute13]
ok: [compute12]
ok: [compute8]
ok: [compute7]
ok: [compute16]
ok: [compute15]
ok: [compute14]
ok: [compute17]
ok: [compute10]
ok: [compute18]
```

after run, on a compute box

```
nova@573960-compute12:~$ cat .ssh/
config       id_rsa       id_rsa.pub   known_hosts
```

**Comment by andymcc** [comment]
@jacobwagner can you rerun the nova-compute play with increased verbosity to see what its doing on the authorized keys section?

I've literally just done a clean run now (based off the ansible-lxc-rpc repo) and it worked ok for me on both my compute nodes:

root@node11:~# ls /var/lib/nova/.ssh/
authorized_keys  config  id_rsa  id_rsa.pub

root@node15:~# ls /var/lib/nova/.ssh/
authorized_keys  config  id_rsa  id_rsa.pub

**Comment by andymcc** [comment]
Since the id_rsa.pub file is present, its odd that the authorized_keys file wasn't created. All the authorized_keys task does is copy that using the "authorized_key" ansible module.

**Comment by andymcc** [comment]
I've made a "second" patch (https://github.com/andymcc/ansible-lxc-rpc/commit/b71621343a78f60231c5d3499ec5f2abd562747b) which just does a straight copy instead of using the "authorized_key" ansible module so if you are still having trouble that will work, but it would definitely be better to figure out why this didn't work for you - I can see all the appropriate files are created on the host which you ran the ansible playbooks from and copied up to the compute nodes - so the only task that can be failing is the "Add key to authorized_keys file"

**Comment by jacobwagner** [comment]
fix in PR #72, closing
